File: Anotacoes_de_sala/20220823 Curso de C - Parte 2/001 Raiz quadrada Heron/raiz-quadrada-busca-binaria.c
#include <stdio.h>
#include <math.h>
#define ERROR 1e-12

double square_root(double x) {
	double low = 0, high = x, guess = x;
	while (fabs(guess * guess - x) > ERROR) {
		if (guess * guess > x)
			high = guess;
		if (guess * guess < x)
			low = guess;
		guess = (low + high) / 2;
	}
	return guess;
}

int main(void) {
	double x;
	printf("Insira um número: ");
	scanf("%lf", &x);
	printf("Raiz quadrada: %g", square_root(x));
	return 0;
}

File: Anotacoes_de_sala/20220823 Curso de C - Parte 2/001 Raiz quadrada Heron/raiz-quadrada-busca-binaria.py
ERROR = 1e-12


def square_root(x):
    low = 0
    high = x
    guess = x
    while not abs(guess * guess - x) <= ERROR:
        if guess * guess > x:
            high = guess
        if guess * guess < x:
            low = guess
        guess = (high + low) / 2
    return guess


def main():
    x = int(input("Insira um número: "))
    print("Raiz quadrada:", square_root(x))


if __name__ == "__main__":
    main()
